Initial preparations for a release

attente opened the pull request.

Commit:
Add .gitignore

Commit:
Check error after launching URL

Commit:
More error checking for invalid URLs

Commit:
Use autotools

allisonkarlitskaya committed:
Add xdg-open

Commit:
Consolidate name to snapd-xdg-open

Commit:
Add license headers

mvo5 commented on src/snapd-xdg-open.c:
```diff
@@ -6,7 +25,7 @@ static GDBusNodeInfo *introspection_data = NULL;
 static const gchar introspection_xml[] =
   "<node>"
-  "  <interface name='com.canonical.SafeLauncher'>"
+  "  <interface name='com.canonical.SnapdXDGOpen'>"
```
This is a bit problematic, the ubuntu-core snap uses the previous (SafeLauncher) name. So if this lands it won't be compatible with the current snapd.

mvo5 commented:
Good stuff! One issue around the name change though.

mvo5 commented:
If the name changes, maybe we can provide the old name as a compat option ?

Commit:
Retain old com.canonical.SafeLauncher bus name and interface

These names are already used in the OS snap and can't be changed easily.

attente commented:
Thanks! Reverted the name change for the bus service and interface for compatibility with the OS snap.

In the future if there's a need to transition to a new name, we can just own the new name and handle the new interface.

Commit:
Run dh_make

The full command used was:
DEBFULLNAME='Ubuntu Desktop Team' \
DEBEMAIL='ubuntu-desktop@lists.ubuntu.com' \
dh_make -p snapd-xdg-open_0.0.0 -s -e ubuntu-desktop@lists.ubuntu.com -c gpl3 --createorig

Commit:
Fill in debian/changelog

Commit:
Fix debian/changelog

Commit:
Bump debian/control standards version

Commit:
Fill in debian/copyright

Commit:
Fill in debian/README.Debian

Commit:
Remove debian templates

Commit:
Correct service name in service file

Commit:
Update .gitignore to ignore debian output

Commit:
Use autoreconf

mvo5 commented on debian/README.source:
```diff
@@ -0,0 +1,10 @@
+------------------------
+
+<this file describes information about the source package, see Debian policy
+manual section 4.14. You WILL either need to modify or delete this file>
```
:-D `You WILL either need to modify or delete this file`

Commit:
Fill in debian/README.source

mvo5 commented on debian/outfile:
```diff
@@ -0,0 +1 @@
+3.0 (quilt)
```
Where does `debian/outfile` come from? Shouldn't that be `debian/source/format` ?

attente commented:
Thanks! Totally missed that...

mvo5 commented:
Looks fine overall, just some tiny nitpicks :)

Commit:
Remove debian/outfile

Commit:
Add debian/source/format
